app/src/main/java/uk/ac/ebi/atlas/experimentpage/cellplot/CellPlotService.java

```diff
@@ -90,4 +93,32 @@ public ImmutableSet<TSnePoint> expressionPlot(String experimentAccession,
                                         pointDto.name()))
                 .collect(toImmutableSet());
     }
+
+    public ImmutableMap<String, JsonObject> fetchDefaultPlotMethodWithParameterisation(String experimentAccession) {
+        var umap = "umap";
+        var tsne = "tsne";
+        var daoResult = cellPlotDao.fetchDefaultPlotMethodWithParameterisation(experimentAccession);
+
+        ImmutableMap.Builder<String, JsonObject> defaultPlotTypeAndOptions = new ImmutableMap.Builder<>();
+        if (!daoResult.isEmpty()) {
+            var umapOptionObject = getMiddleElement(daoResult.get(umap));
+            var tsneOptionObject = getMiddleElement(daoResult.get(tsne));
+
+            defaultPlotTypeAndOptions.put(umap, umapOptionObject);
+            defaultPlotTypeAndOptions.put(tsne, tsneOptionObject);
+        }
+        return defaultPlotTypeAndOptions.build();
+    }
+
+    private static JsonObject getMiddleElement(List plotOptions) {
+
+        if (plotOptions.size() % 2 == 0) { // even number
+            Object umapEvenItem = plotOptions.get((plotOptions.size() / 2 - 1));
```

Comment:
The variable name could be more general for both umap and tsne result.

Reply:
I optimised the code a bit. please have a look.